src/libs/actions/Report.js

```diff
@@ -1205,6 +1205,32 @@ function saveReportActionDraftNumberOfLines(reportID, reportActionID, numberOfLi
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT_NUMBER_OF_LINES}${reportID}_${reportActionID}`, numberOfLines);
 }
 
+/**
+ * @param {String} reportID
+ * @param {String} previousValue
+ * @param {String} newValue
+ */
+function updateNotificationPreference(reportID, previousValue, newValue) {
+    if (previousValue === newValue) {
+        return;
+    }
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {notificationPreference: newValue},
+        },
+    ];
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {notificationPreference: previousValue},
+        },
+    ];
+    API.write('UpdateReportNotificationPreference', {reportID, notificationPreference: newValue}, {optimisticData, failureData});
```

Comment:
@srikarparsi The function is pretty similar to `updateNotificationPreferenceAndNavigate`, I think we can refactor it and reuse it between this function.